### PR TITLE
[native] Fix peakNodeTotalMemoryReservationInBytes report in TaskStatus

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Currently peakNodeTotalMemoryReservationInBytes is set to query task's peak memory reservation
which should be the query pool's peak memory reservation

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/PrestoTask.cpp [hunk 1]
```diff
@@ -356,11 +356,11 @@ protocol::TaskStatus PrestoTask::updateStatusLocked() {
...
   }
 
-  const auto veloxMemStats = task->pool()->stats();
-  info.taskStatus.memoryReservationInBytes = veloxMemStats.currentBytes;
+  const auto veloxTaskMemStats = task->pool()->stats();
+  info.taskStatus.memoryReservationInBytes = veloxTaskMemStats.currentBytes;
   info.taskStatus.systemMemoryReservationInBytes = 0;
   info.taskStatus.peakNodeTotalMemoryReservationInBytes =
-      veloxMemStats.peakBytes;
+      task->queryCtx()->pool()->peakBytes();
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/PrestoTask.cpp:363]: Perhaps, add a comment to explain that a given worker may run multiple tasks from the same query. peakNodeTotalMemoryReservationInBytes represents peak memory usage across all these tasks.

File: presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp [hunk 2]
```diff
@@ -1004,11 +1004,37 @@ TEST_F(TaskManagerTest, distributedSort) {
...
+  const auto finalPrestoTaskStatus =
+      taskManager_
+          ->getTaskStatus(
+              finalTaskId, std::nullopt, std::nullopt, statusRequestState)
+          .wait()
+          .get();
+  ASSERT_EQ(
+      finalPrestoTaskStatus->peakNodeTotalMemoryReservationInBytes,
+      finalPrestoTaskStats.peakNodeTotalMemoryInBytes);
+  // Since we have more than one tasks in a query, then the task peak memory
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp:1030]: The comment suggests that the assert should be <=, not ==.

**Review by mbasmanova — DISMISSED** [review]
Thanks.

File: presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp:1030]: Yeah, the test failed. Thanks and updated.

**Review by tanjialiang — APPROVED** [review]

**Comment by tanjialiang** [comment]
Thanks for fixing it

**Pull request merged**